datalad/utils.py

```diff
@@ -610,7 +654,15 @@ def rmtemp(f, *args, **kwargs):
         lgr.info("Keeping temp file: %s", f)
 
 
-def file_basename(name, return_ext=False):
+@overload
+def file_basename(name: str | Path, return_ext: Literal[True]) -> tuple[str, str]:
+    ...
+
+@overload
+def file_basename(name: str | Path, return_ext: Literal[False] = False) -> str:
+    ...
+
+def file_basename(name: str | Path, return_ext: bool =False) -> str | tuple[str, str]:
```

Comment:
Due to use of more specific `@overload`s

```suggestion
def file_basename(name, return_ext=False):
```